DPPP/MSBDAReader.cc

```diff
@@ -185,12 +202,14 @@ bool MSBDAReader::process(const DPBuffer&) {
       continue;
     }
 
-    size_t blNr = bl_to_id_[std::make_pair(ant1_col(nread_), ant2_col(nread_))];
+    const auto ant12 = std::make_pair(ant1_col(nread_), ant2_col(nread_));
 
-    buffer->AddRow(ms_time, interval[i], exposure[i], blNr,
-                   desc_id_to_nchan_[data_desc_id[i]], info().ncorr(),
-                   read_vis_data_ ? data[i].data() : nullptr, nullptr,
-                   weights[i].data(), nullptr, uvw[i].data());
+    if (!buffer->AddRow(ms_time, interval[i], exposure[i], ant_to_bl_[ant12],
+                        desc_id_to_nchan_[data_desc_id[i]], info().ncorr(),
+                        read_vis_data_ ? data[i].data() : nullptr, nullptr,
+                        weights[i].data(), nullptr, uvw[i].data())) {
+      throw std::runtime_error("Internal MSBDAReader error: Buffer too small");
```

Comment:
I'd prefer a different kind of error message. This will only occur if we (programmers) have made a mistake and a user will never be able to fix this. Maybe change it that they should contact the DP3 team or something. 

I see this occurring when info.nchan() is something like 2 (if you take average instead of max to determine buffer size). And then you get buffers with a channel width of 4. This exception will occur.